src/block/DragonEgg.php

```diff
@@ -57,7 +58,9 @@ public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player
 	}
 
 	public function onAttack(Item $item, int $face, ?Player $player = null) : bool{
-		$this->teleport();
+		if($player !== null && $player->getGamemode() != GameMode::CREATIVE()){
```

Comment:
```suggestion
		if($player !== null && !$player->getGamemode()->equals(GameMode::CREATIVE())){
```

Reply:
Is there a reason why !== is not reliable here?

Reply:
Threads make things complicated. The enum members are able to be cloned and serialized.